src/main/java/eu/bbmri_eric/negotiator/database/model/Network.java

```diff
@@ -0,0 +1,132 @@
+package eu.bbmri_eric.negotiator.database.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString.Exclude;
+
+/**
+ * Represents a Network entity in the database. A Network has a unique identifier, a URI, a name, a
+ * contact email, a set of managers, and a set of resources.
+ */
+@Getter
+@Setter
+@Entity
+@Builder
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class Network {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "network_id_seq")
+  @SequenceGenerator(name = "network_id_seq", initialValue = 10000, allocationSize = 1)
+  private Long id;
+
+  /** The URI of the network */
+  @NotNull private String uri;
+
+  /** The name of the network */
+  @Column(unique = true)
+  private String name;
+
+  /** A unique and persistent identifier issued by an appropriate institution */
+  @NotNull
+  @Column(unique = true)
+  private String externalId;
+
+  /** The contact email of the network. */
+  private String contactEmail;
+
+  /** The managers of the network. */
+  @ManyToMany(mappedBy = "networks")
+  @Exclude
+  @Setter(AccessLevel.NONE)
+  @Builder.Default
+  private Set<Person> managers = new HashSet<>();
+
+  /** The resources of the network. */
+  @ManyToMany(
+      cascade = {CascadeType.PERSIST, CascadeType.MERGE},
+      fetch = FetchType.EAGER)
```

Comment:
I would avoid eager fetching as much as possible since we will have some large objects. Any good reason for including it? A network can have thousands of resources